### PR TITLE
Implementation of factories to provide Apache HTTP Clients

### PR DESCRIPTION
**Pull request by ryanjbaxter**
Adds factory beans for downstream projects to consistently create Apache HTTP Clients

### CODE REVIEW

File: spring-cloud-commons/src/main/java/org/springframework/cloud/commons/httpclient/DefaultOkHttpClientFactory.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+/**
+ * Default implementation of {@link OkHttpClientFactory}.
+ * @author Ryan Baxter
+ */
+public class DefaultOkHttpClientFactory implements OkHttpClientFactory {
+
+	private static final Log LOG = LogFactory.getLog(DefaultOkHttpClientFactory.class);
+
+	@Override
+	public OkHttpClient create(boolean disableSslValidation, long connectTimeout,
```
Comment (spencergibb) [inline spring-cloud-commons/src/main/java/org/springframework/cloud/commons/httpclient/DefaultOkHttpClientFactory.java]: I'm a little worried about such a long list of parameters. I wonder if something like a builder customizer might make more sense?

**Comment by ryanjbaxter** [comment]
@spencergibb I ended up just using the provided builders for the clients instead of reinventing the wheel.  I left the connection pools as is since the number of arguments is rather fixed.

**Comment by ryanjbaxter** [comment]
@spencergibb want to take another pass at this?

**Comment by ryanjbaxter** [comment]
ping @spencergibb

**Pull request merged**